develop/smart-contracts/dev-environments/hardhat.md

```diff
@@ -80,129 +88,129 @@ The plugin will compile your Solidity contracts for Solidity versions `0.8.0` an
 
-!!! warning "Contract Size Limitation in Testing Environment"
+To run your test:
 
-    When deploying large contracts, you might encounter: `Error: the initcode size of this transaction is too large`.
+1.  Update the `hardhat.config.js` file accordingly to the [Testing Environments](#testing-environments) section
 
-    This limitation is imposed by Hardhat's client-side checks, not by PolkaVM itself. As a workaround, you can use a direct `JsonRpcProvider`:
+2.  Execute the following command:
 
-    ```javascript
-    --8<-- "code/develop/smart-contracts/dev-environments/hardhat/disclaimer-json-rpc-provider-alternative.js"
+    ```bash
+    npx hardhat test
     ```
 
-    For more details, see [this GitHub issue](https://github.com/paritytech/contract-issues/issues/47#issuecomment-2790181622){target=\_blank}.
-
-
-
-1. Update the Hardhat configuration file to add the local network as a target for local deployment:
+## Deploying with a Local Node
 
-    === "npm Configuration"
+Before deploying to a live network, you can deploy your contract to a local node using [Ignition](https://hardhat.org/ignition/docs/getting-started#overview){target=\_blank} modules:
 
-        ```javascript title="hardhat.config.js" hl_lines="34-37"
-        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:1:21'
-          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:37:53'
-        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:59:60'
-        ```
+1.  Update the Hardhat configuration file to add the local network as a target for local deployment:
 
-    === "Binary Configuration"
+    ```javascript title="hardhat.config.js" hl_lines="12-15"
+    --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:1:4'
 
-        ```javascript title="hardhat.config.js" hl_lines="36-39"
-        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:1:8'
-          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:22:53'
-        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:59:60'
-        ```
+    --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:6:7'
+        ...
+        --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:12:13'
+            ...
+          --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:24:28'
+    --8<-- 'code/develop/smart-contracts/dev-environments/hardhat/hardhat.config.js:33:35'
+    ```
 
-2. Start a local node:
+2.  Start a local node:
 
     ```bash
     npx hardhat node
     ```
 
     This command will spawn a local substrate node along with the ETH-RPC adapter.
 
-3. In a new terminal window, deploy the contract using Ignition:
+3.  In a new terminal window, deploy the contract using Ignition:
```

Comment:
All these spaces should be removed. Not sure if you run prettier on the file? But it seems like some sort of formatter is going in and causing chaos